tests/test_metrics_calibration.py

```diff
@@ -31,12 +33,108 @@ def test_sharpness_on_test_set(supply_test_set):
     assert np.abs(sharpness(test_std) - 0.648074069840786) < 1e-6
 
 
-def test_get_proportion_lists_on_test_set(supply_test_set):
-    """Test get_proportion_lists on the test set for some dummy values."""
-    test_exp_props, test_obs_props = get_proportion_lists(
-        *supply_test_set, num_bins=100, recal_model=None
+def test_root_mean_squared_calibration_error_on_test_set(supply_test_set):
+    """Test root mean squared calibration error on some dummy values."""
+    test_rmsce_nonvectorized = root_mean_squared_calibration_error(
+        *supply_test_set,
+        num_bins=100,
+        vectorized=False,
+        recal_model=None,
+        prop_type="interval"
     )
-    assert len(test_exp_props) == len(test_obs_props)
+    test_rmsce_vectorized = root_mean_squared_calibration_error(
+        *supply_test_set,
+        num_bins=100,
+        vectorized=True,
+        recal_model=None,
+        prop_type="interval"
+    )
+    assert np.abs(test_rmsce_nonvectorized - test_rmsce_vectorized) < 1e-6
+    assert np.abs(test_rmsce_vectorized - 0.4165757476562379) < 1e-6
+
+
+def test_mean_absolute_calibration_error_on_test_set(supply_test_set):
+    """Test mean absolute calibration error on some dummy values."""
+    test_mace_nonvectorized = mean_absolute_calibration_error(
+        *supply_test_set,
+        num_bins=100,
+        vectorized=False,
+        recal_model=None,
+        prop_type="interval"
+    )
+    test_mace_vectorized = mean_absolute_calibration_error(
+        *supply_test_set,
+        num_bins=100,
+        vectorized=True,
+        recal_model=None,
+        prop_type="interval"
+    )
+    assert np.abs(test_mace_nonvectorized - test_mace_vectorized) < 1e-6
+    assert np.abs(test_mace_vectorized - 0.3733333333333335) < 1e-6
+
+
+def test_adversarial_group_calibration_on_test_set(supply_test_set):
+    """Test adversarial group calibration on test set for some dummy values."""
+    test_out = adversarial_group_calibration(
+        *supply_test_set,
+        cali_type="mean_abs",
+        prop_type="interval",
+        num_bins=100,
+        num_group_bins=10,
+        draw_with_replacement=False,
+        num_trials=10,
+        num_group_draws=10,
+        verbose=False
+    )
+
+    assert np.max(np.abs(test_out.group_size - np.linspace(0, 1, 10))) < 1e-6
+    assert test_out.score_mean[0] < 0.5
+    assert np.abs(test_out.score_mean[-1] - 0.37333333) < 1e-6
+    assert np.min(test_out.score_stderr) >= 0
+
+
+def test_miscalibration_area_on_test_set(supply_test_set):
+    """Test miscalibration area on some dummy values."""
+    test_miscal_area_nonvectorized = miscalibration_area(
+        *supply_test_set,
+        num_bins=100,
+        vectorized=False,
+        recal_model=None,
+        prop_type="interval"
+    )
+    test_miscal_area_vectorized = miscalibration_area(
+        *supply_test_set,
+        num_bins=100,
+        vectorized=True,
+        recal_model=None,
+        prop_type="interval"
```

Comment:
Do you think we also need a test for miscalibration area on `prop_type="quantile"`?